modules/exploits/windows/http/prtg_authenticated_rce_cve_2023_32781.rb

```diff
@@ -0,0 +1,301 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PRTG CVE-2023-32781 Authenticated RCE UPDATED',
+        'Description' => %q{
+          Authenticated RCE in Paessler PRTG
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Kevin Joensen <kevin[at]baldur.dk>'],
+        'References' => [
+          [ 'URL', 'https://baldur.dk/blog/prtg-rce.html'],
+          [ 'CVE', '2023-32781']
+        ],
+        'DisclosureDate' => '2023-08-09',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          [
+            'Windows',
+            {
+              'Arch' => [ ARCH_X64, ARCH_X86 ],
+              'Platform' => 'win',
+              'CmdStagerFlavor' => [ 'psh_invokewebrequest' ]
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'WfsDelay' => 0 # Schedules takes some time to run
+        },
```

Comment:
Setting `WfsDelay` to 0 is not going to cause this to run indefinitely. If you need to provide it with more time, you might want to increase it to 30. If it's in the minutes or hours, you may need to change the stance to passive.

Reply:
Just removed `WfsDelay` as it is not used. The exploit is instant.